app/src/main/java/com/sdp/movemeet/backend/firebase/firestore/FirestoreManager.java

```diff
@@ -49,22 +50,33 @@ public FirestoreManager(FirebaseFirestore db, String collection, BackendSerializ
     @Override
     public Task<Void> add(T object, String path) {
         if (object == null) throw new IllegalArgumentException();
-
-        String documentPath = db.collection(collection).document().getPath();
+        String documentPath;
+        if (path == null) {
+            documentPath = db.collection(collection).document().getPath();
+        } else {
+            documentPath = path;
+        }
         object.setDocumentPath(documentPath);
         Map<String, Object> data = serializer.serialize(object);
 
         return db.document(object.getDocumentPath()).set(data);
     }
 
     @Override
-    public Task<Void> set(T object, String path, String field, String value) {
+    public Task<Void> set(T object, String path) {
         if (object == null) throw new IllegalArgumentException();
-        // TODO: check why the number of registered participants vary on Firebase Firestore
         Map<String, Object> data = serializer.serialize(object);
+
         return db.document(path).set(data);
     }
 
+    @Override
+    public Task<Void> updt(String value, String path, String field) {
+        if (value == null || path == null) throw new IllegalArgumentException();
```

Comment:
Can't the field be null ?

Reply:
You're right, it shouldn't be `null`, I updated this method accordingly this way:

```java
@Override
    public Task<Void> update(String path, String field, String value) {
        if (path == null || field == null || value == null) throw new IllegalArgumentException();

        return db.document(path).update(field, FieldValue.arrayUnion(value)); // object will be in this case the "String value" containing the ID of the participant to add!
    }
```